### PR TITLE
Add missing tests for ReadableStream

### PR DESCRIPTION
**Pull request by ricea**
* Test the specific rejection returned by calling it.return() twice (to avoid
  being confused by assertions).
* Verify that it.next() after it.return() fails.
* Verify that it.return() unlocks the stream synchronously.
* Add brand checks for getIterator(), locked and desiredSize.
* Verify that canceling tee branches in reverse order works.
* Test that erroring a teed stream errors both branches, even when they
  are not waiting for chunks.
* Verify that controller.close() clears the list of pending reads

### CODE REVIEW

File: streams/readable-streams/async-iterator.any.js [hunk 1]
```diff
@@ -338,3 +335,10 @@ promise_test(async () => {
   assert_equals(readResult.value, 3, 'should read remaining chunk');
   await reader.closed;
 }, 'Acquiring a reader and reading the remaining chunks after partially async-iterating a stream with preventCancel = true');
+
+promise_test(async t => {
+  const rs = new ReadableStream();
+  const it = rs.getIterator();
+  await it.return();
+  return promise_rejects(t, new TypeError(), it.next(), 'next() should reject');
+}, 'calling next() after return() should reject');
```
Comment (domenic) [inline streams/readable-streams/async-iterator.any.js]: Should we also test what happens if you do these two tests (return-return and return-next) without awaiting the return? I think that is also determined, right?

File: streams/readable-streams/tee.any.js [hunk 2]
```diff
@@ -184,6 +185,32 @@ promise_test(() => {
 
 }, 'ReadableStream teeing: canceling both branches should aggregate the cancel reasons into an array');
 
+promise_test(() => {
+
+  const reason1 = new Error('This little one\'s not worth the effort.');
+  const reason2 = new Error('Come, let me get you something.');
```
Comment (domenic) [inline streams/readable-streams/tee.any.js]: 💯

**Review by domenic — APPROVED** [review]
LGTM with potential addition

File: streams/readable-streams/async-iterator.any.js (hunk 1 above)
Comment (ricea) [reply streams/readable-streams/async-iterator.any.js]: Instead I added a test that the lock is released synchronously by return(), which is roughly equivalent and slightly improves coverage.

**Pull request merged**